src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/extension.ts

```diff
@@ -119,44 +127,55 @@ export async function activate(context: ExtensionContext, languageServerDir: str
                 documentManager,
                 languageServiceClient,
                 logger);
+            const semanticTokenProvider = new RazorDocumentSemanticTokensProvider(
+                documentSynchronizer,
+                documentManager,
+                languageServiceClient,
+                logger);
 
             localRegistrations.push(
                 languageConfiguration.register(),
                 provisionalCompletionOrchestrator.register(),
-                vscode.languages.registerCodeActionsProvider(
+                vscodeType.languages.registerCodeActionsProvider(
                     RazorLanguage.id,
                     codeActionProvider),
-                vscode.languages.registerCompletionItemProvider(
+                vscodeType.languages.registerCompletionItemProvider(
                     RazorLanguage.id,
                     completionItemProvider,
                     '.', '<', '@'),
-                vscode.languages.registerSignatureHelpProvider(
+                vscodeType.languages.registerSignatureHelpProvider(
                     RazorLanguage.id,
                     signatureHelpProvider,
                     '(', ','),
-                vscode.languages.registerDefinitionProvider(
+                vscodeType.languages.registerDefinitionProvider(
                     RazorLanguage.id,
                     definitionProvider),
-                vscode.languages.registerImplementationProvider(
+                vscodeType.languages.registerImplementationProvider(
                     RazorLanguage.id,
                     implementationProvider),
-                vscode.languages.registerHoverProvider(
+                vscodeType.languages.registerHoverProvider(
                     RazorLanguage.documentSelector,
                     hoverProvider),
-                vscode.languages.registerReferenceProvider(
+                vscodeType.languages.registerReferenceProvider(
                     RazorLanguage.id,
                     referenceProvider),
-                vscode.languages.registerCodeLensProvider(
+                vscodeType.languages.registerCodeLensProvider(
                     RazorLanguage.id,
                     codeLensProvider),
-                vscode.languages.registerRenameProvider(
+                vscodeType.languages.registerRenameProvider(
                     RazorLanguage.id,
                     renameProvider),
                 documentManager.register(),
                 csharpFeature.register(),
                 htmlFeature.register(),
                 documentSynchronizer.register(),
                 reportIssueCommand.register());
+            if (legend) {
+                localRegistrations.push(vscodeType.languages.registerDocumentSemanticTokensProvider(
```

Comment:
We need to do the equivilent of "ifdef" on this, but typescript doesn't have functionality like that.